cpp/src/io/orc/reader_impl_helpers.hpp

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright (c) 2019-2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "aggregate_orc_metadata.hpp"
+#include "orc.hpp"
+
+#include <io/utilities/column_buffer.hpp>
+
+#include <cudf/io/orc.hpp>
+
+#include <rmm/cuda_stream_view.hpp>
+
+#include <memory>
+#include <vector>
+
+namespace cudf::io::orc::detail {
+using namespace cudf::io::detail;
+
+/**
+ * @brief Keeps track of orc mapping and child column details.
+ */
+struct reader_column_meta {
+  // Mapping between column id in orc to processing order.
+  std::vector<std::vector<size_type>> orc_col_map;
+
+  // Number of rows in child columns.
+  std::vector<uint32_t> num_child_rows;
```

Comment:
A lot of `uint32_t`s used instead of `cudf::size_type` in these files. Not a suggestion for this PR, just a note for potential future clean up.